### ISSUE TITLE
[BUG] Missing "results" field in SARIF reports for valid resources

### ISSUE THREAD

**Issue by rm3l** [stackrox/kube-linter#213]
**System info:**
```
 ❯ kube-linter version
development
```
 - Ubuntu Linux 21.04

**Describe the bug**
Hi there!
After fixing the warnings reported by KubeLinter, I tried to upload the generated SARIF reports to GitHub Code Scanning, and got the following errors by the `github/codeql-action/upload-sarif` Action:
```
Error: Invalid SARIF. Missing 'results' array in run.
      at countResultsInSarif (/home/runner/work/_actions/github/codeql-action/v1/lib/upload-lib.js:173:19)
      at uploadFiles (/home/runner/work/_actions/github/codeql-action/v1/lib/upload-lib.js:270:30)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Object.uploadFromActions (/home/runner/work/_actions/github/codeql-action/v1/lib/upload-lib.js:132:12)
      at async run (/home/runner/work/_actions/github/codeql-action/v1/lib/upload-sarif-action.js:49:29)
      at async runWrapper (/home/runner/work/_actions/github/codeql-action/v1/lib/upload-sarif-action.js:61:9)
```

**To Reproduce**
Just run `kube-linter` against the sample YAML file below.

With the default `plain` output format, we can confirm no error is reported by KubeLinter:
```sh
❯ kube-linter lint my-deployment.yaml --format plain                                       
KubeLinter development

No lint errors found!
```

However, when requesting a SARIF output format, the JSON returned is missing the `runs > results` field, which makes this not valid, I guess, per the specifications. This field is indeed required by GitHub Code Scanning, as documented here : https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#run-object

```sh
❯ kube-linter lint my-deployment.yaml --format sarif | jq '.runs[].results[]'
jq: error (at <stdin>:0): Cannot iterate over null (null)
```

**Sample YAML input**
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: my-webserver
  name: my-webserver
  namespace: my-ns
spec:
  replicas: 1
  selector:
    matchLabels:
      app: my-webserver
  strategy: {}
  template:
    metadata:
      labels:
        app: my-webserver
    spec:
      securityContext:
        runAsNonRoot: true
      containers:
      - image: nginx:alpine
        name: nginx
        resources:
          limits:
            memory: 100Mi
            cpu: 500m
          requests:
            memory: 50Mi
            cpu: 250m
        securityContext:
          readOnlyRootFilesystem: true
```

**Expected behavior**
Per the [SARIF specifications](https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html#_Toc10127698):

> If the tool failed to start, and if the engineering system responsible for running the tool synthesized a SARIF file to record the failure, then results MAY be present. If it is present, its value SHALL be null. See §3.20.13, invocation.processStartFailureMessage, for more about this scenario.
> 
> If the tool started but failed to begin its analysis (for example, because its command line was invalid), then again results MAY be present, and if present SHALL be null.
> 
> In all other circumstances, results SHALL be present and SHALL contain all results detected by the tool. If the tool did not detect any results, results SHALL be an empty array.
> 
> If results is absent, it SHALL default to null.

So per this sentence (`If the tool did not detect any results, results SHALL be an empty array.`), I would expect `results` to be an empty array rather than just missing.

**Additional context**
As expected, the `results` field is actually present if KubeLinter detects errors in the Kubernetes YAML file, and this integrates very well with GitHub Code Scanning Alerts.

**Comment by msugakov** [comment]
Hi @rm3l 
Thanks for reporting this! This is indeed a bug. I patched it in the development version.
I'll let you know when kube-linter with this fix will be released.

**Comment by rm3l** [comment]
Great — thanks @msugakov !

**Comment by rm3l** [comment]
@msugakov I just tested the latest development version (`v0.0.0-20210920192959-c12af3736628`), and I can confirm your changes in #214 fixed the issue. Thanks.

**Comment by viswajithiii** [comment]
Thanks again @rm3l for filing this issue! We've now released 0.2.4 of kube-linter which includes this fix.

**Comment by rm3l** [comment]
Thank you guys for your prompt fixes!
